tests/48admin.pl

```diff
@@ -273,6 +273,12 @@ sub await_purge_complete {
       matrix_invite_user_to_room_synced( $user, $remote_user, $room_id )
       ->then( sub {
          matrix_join_room_synced( $remote_user, $room_id )
+      })->then( sub {
+         # Force the remote server (`server-1`) to backfill all of the events to the
+         # beginning of the room. We want to make sure to de-outlier all of the state
+         # (like the invite event) before the join so it's available later when
+         # `server-0` asks about it again.
+         matrix_get_room_messages( $remote_user, $room_id, limit => 100 )
```

Comment:
Closing as there may be a better way forward, https://github.com/matrix-org/synapse/pull/13816#discussion_r987711019